programming_exercises/submit_from_git.rst

```diff
@@ -0,0 +1,54 @@
+Submit from Git
+===============
+
+A-plus supports exercises where student solutions are retrieved
+from their personal Git repository.
+
+.. warning::
+  
+  To test grading from Git, **a private RSA key MUST be appened to
+  the file at ``exercises/git_hello_python/git_rsa``.**
+  In May 2021, the access script in `grading-base`_ requires RSA key
+  that is used in cloning the repository via ssh URL.
+  Furthermore, it may require the repository is in `version.aalto.fi`_.
+
+  Inserting the private key has the consequence that any users having
+  access to these course source files can run git commands in the system
+  authenticated as the user whose account is linked to the private key
+  via the matching RSA public key configured in GitLab.
```

Comment:
Is it expected here that the teacher would add their own personal private key to the repository? If that's the case, it might be worth considering course-specific [Deploy keys](https://version.aalto.fi/gitlab/help/ssh/README#per-repository-deploy-keys) as an alternative, depending on the actual use case.

Reply:
Yes, this dates back to times of two A+ courses running on a single virtual machine. Deploy keys should work but students cannot easily append the same key (current gitlab versions allow to add a deploy key from another project you own). One solution is for the students to add a teacher/TA as a maintainer who will then add the same deploy key to every student project. A best strategy might be that students can configure their personal course repository and required keys into A+ (related to course enrolment model) so that they can just click grade-button once they have setup their course. If such development is considered then it might be worthwhile to support a student choice whether they want to submit from git or upload separate files.